test/integration/testImage/src/tarResults.go

```diff
@@ -64,3 +68,78 @@ func reportTarFile(cmd *cobra.Command, args []string) error {
 	err = ioutil.WriteFile(doneFile, []byte(tb), os.FileMode(0666))
 	return errors.Wrap(err, "failed to write to done file")
 }
+
+// tarDir tars up an entire directory and outputs the tarball to the specified output path.
+// If useGzip is true, it also gzips the resulting tarball.
+// NOTE: Copied from sonobuoy itself; duplicated since this is a separate binary/project
+// so it messed with the build when introducing this method the first time.
+func tarDir(dir, outpath string, useGzip bool) error {
```

Comment:
Would it be better to make `tarDir` exportable in its own package as `TarDir` then reuse it in both places?

Reply:
So it would have to be outside of sonobuoy entirely and outside of testdir.

I was tempted to make another repo entirely in my own github name but just thought this was easier for now. It does introduce an issue with keeping them in sync though.

Would you prefer if I made a tiny repo for tar/untar? Effectively a super simple version of tarx?

Reply:
@johnSchnake maybe I misunderstood the issue. But it seems that if sonobuoy repo is viewed as a monorepo, but generates multiple binaries, you should be able to reuse the `TarDir` (from it's own package).   

Unless `test` is treated as a completely separate Go project, in that case it should have its own go.mod, then it can import a package with the `TarDir` (I think you can have go.mod in subpackages, could be wrong). 

But please set me straight if I misunderstand the issue.

Reply:
No, thats pretty much the issue but there is just a problem when you simultaneously try and change the sonobuoy repo and the test repo at the same time. So the test binary, when getting built says "oh I need to go get sonobuoy for this tar function..." but then since that isn't anywhere other than my branch/fork it can't find it. So we can depend on sonobuoy just fine, but it would always need to be looking back at older commits. So if we get rid of tarx in sonobuoy and commit it, then have a separate commit where we update the test image to leverage that it would be fine.

We could do that: split it into two commits. We can just copy it here for this commit and follow up with just referencing sonobuoys. That way it is always in sync and after just this one commit I'll follow up and then testimage can depend directly on sonobuoy.